File: Python3/1524. Number of Sub-arrays With Odd Sum.py
class Solution:
    def numOfSubarrays(self, arr: List[int]) -> int:
        MODULO = 10 ** 9 + 7
        prefix = ans = 0

        even, odd = 1, 0

        for num in arr:
            prefix += num
            if prefix % 2:
                ans = (ans + even) % MODULO
                odd += 1
            else:
                ans = (ans + odd) % MODULO
                even += 1

        return ans

File: Java/1524. Number of Sub-arrays With Odd Sum.java
class Solution {
    public int numOfSubarrays(int[] arr) {
        int MODULO = (int)Math.pow(10, 9) + 7;
        int even = 1, odd = 0, prefix = 0, ans = 0;

        for (int num: arr) {
            prefix += num;
            if (prefix % 2 == 1) {
                ans = (ans + even) % MODULO;
                odd++;
            } else {
                ans = (ans + odd) % MODULO;
                even++;
            }
        }

        return ans;
    }
}